lifecycle/services/annotate.go

```diff
@@ -0,0 +1,207 @@
+package services
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/jfrog/jfrog-client-go/utils"
+	"github.com/jfrog/jfrog-client-go/utils/distribution"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"net/http"
+	"path"
+	"strings"
+)
+
+const (
+	Tag               = "tag"
+	PropertiesBaseApi = "api/storage"
+)
+
+type CommonPropParams struct {
+	Path      string `json:"path"`
+	Recursive bool   `json:"recursive"`
+}
+
+type ArtCommonParams struct {
+	Url string `json:"url"`
+}
+
+type AnnotateOperationParams struct {
+	RbTag          RbAnnotationTag
+	RbProps        RbAnnotationProps
+	RbDelProps     RbDelProps
+	RbDetails      ReleaseBundleDetails
+	QueryParams    CommonOptionalQueryParams
+	PropertyParams CommonPropParams
+	ArtifactoryUrl ArtCommonParams
+}
+
+type RbDelProps struct {
+	Keys  string `json:"keys"`
+	Exist bool   `json:"exist"`
+}
+
+type RbAnnotationTag struct {
+	Tag   string `json:"tag,omitempty"`
+	Exist bool   `json:"exist"`
+}
+
+type RbAnnotationProps struct {
+	Properties map[string][]string `json:"properties,omitempty"`
+	Exist      bool                `json:"exist"`
+}
+
+func (rbs *ReleaseBundlesService) AnnotateReleaseBundle(params AnnotateOperationParams) error {
+	return rbs.annotateReleaseBundle(params)
+}
+
+func GetReleaseBundleSetTagApi(rbDetails ReleaseBundleDetails) string {
+	return path.Join(releaseBundleBaseApi, records, rbDetails.ReleaseBundleName, rbDetails.ReleaseBundleVersion, Tag)
+}
+
+func (rbs *ReleaseBundlesService) setTag(params AnnotateOperationParams, api string, httpClientsDetails httputils.HttpClientDetails) error {
```

Comment:
```suggestion
func (ao *AnnonateOperation) setTag(params AnnotateOperationParams, api string) error {
```
```
type AnnonateOperation struct {
 rtDetails *auth.ServiceDetails
 params AnnonateOperationParams
}
```
may be having a new struct for annotate operation will solve lot of problems with handling api, path, httpClientDetails makes more readable as well.
manager.go can have a new API on lifecycle service which take rtDetails as an input param.
WDYT?

Reply:
@bhanurp I prefer not to change it. Because - 1) I see that in lifecycle/manager all methods are written by the same pattern. So, I'd like to keep it consistent. 2) I don't really know where from I get all needed data when I am in manager.go 3) I don't see any benefits from this change - there are no problems with handling api, path and httpClientDetails 4) We are luck of time, you didn't mention it in your previous code review, so, I suppose it's good enough. 
WDYT?